include/rocksdb/file_system.h

```diff
@@ -254,6 +255,9 @@ struct IODebugContext {
   // Arbitrary structure containing cost information about the IO request
   std::any cost_info;
 
+  // IODebugContext may receive concurrent reads/writes
```

Comment:
If I just look at this class, it seems weird that there is a mutex and no accessor / modifier functions use it. Is this simply a mutex available to FileSystem implementations that they can use to synchronize reads and writes to this IODebugContext?